Sources/DataStructures/BinaryHeap.swift

```diff
@@ -0,0 +1,122 @@
+//
+//  BinaryHeap.swift
+//  PitchSpeller
+//
+//  Created by Benjamin Wetherfield on 7/15/18.
+//
+
+struct BinaryHeap<Element: Hashable, Value: Comparable> {
+
+    // MARK: - Instance Properties
+    
+    private var storage: [Element]
+    private var lookup: [Element: Value]
+    private var indices: [Element: Int]
+    
+    // MARK: - Instance Methods
+    
+    private func lessAt(_ i: Int, than j: Int) -> Bool {
+        return value(at: i) < value(at: j)
+    }
+    
+    mutating func insert (_ element: Element, _ value: Value) {
```

Comment:
I don't dislike the whitespace here (`mutating func insert (`), but we should be consistent.

Reply:
👍